All the global flags have been removed. Standard arguments are now being used instead

matte-esse opened the pull request:
See submodule PR here: https://github.com/jamesdolezal/simclr/pull/1

NB: 

https://github.com/jamesdolezal/slideflow/blob/ece9a685fbf95defcc69d53deee3df9d278d5487/slideflow/model/__init__.py#L384 

and 

https://github.com/jamesdolezal/slideflow/blob/ece9a685fbf95defcc69d53deee3df9d278d5487/slideflow/model/__init__.py#L439

Don't use the arguments/parameters of the actual model, this is the next thing that needs to be discussed and fixed.

matte-esse commented:
I tried to squash all the commits into one but it wouln't let me, I think it's because I messed up the author at first and only later fixed it. @jamesdolezal if you want a single commit for this feature I can create a new branch and have all the edits in a single commit.

jamesdolezal commented:
I've added quite a few changes here and in the simclr submodule. Notable, I switched the flags from a `SimpleNamespace` to a custom class with a method `model_kwargs`, to make it a bit cleaner to pass around those specific keyword arguments. 

This is the code I've been using to test training:

```python
import slideflow as sf
from slideflow import simclr

P = sf.Project('.../path/to/test_project')
dataset = P.dataset(tile_px=96, tile_um='40x').clip(10)

# Test unsupervised training (no labels)
simclr_args = simclr.get_args(image_size=96)
P.train_simclr(
  'test_experiment',
  train_dts=dataset,
  val_dts=dataset,
  simclr_args=simclr_args)

# Test supervised training
simclr_args = simclr.get_args(image_size=96, lineareval_while_pretraining=True)
P.train_simclr(
  'test_experiment',
  train_dts=dataset,
  val_dts=dataset,
  labels='benign_malignant',
  simclr_args=simclr_args)
```

and this is how I've been testing the generation:

```python
import slideflow as sf

P = sf.Project('.../path/to/test_project')
sf.DatasetFeatures(
   '/path/to/simclr/saved_model',
  dataset=P.dataset(tile_px=96, tile_um='40x').clip(10))
```

Although you'd need to change the project path and outcome label to whatever you're using.

matte-esse commented:
It works on my end too. I have fixed a thing that I had pushed by mistake and moved the comments in the keyword arguments of `SimCLR_Args` to the docstring
